### PR TITLE
ci: run the linter from travis with every build

### PR DESCRIPTION
**Pull request by Fishrock123**
It'd be nice to have it run as a separate task, but I think this should
work for now.

Refs: https://github.com/ayojs/ayo/pull/71

cc @addaleax 

##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

ci

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,3 +18,4 @@ install:
   - make -j2 V=
 script:
   - make -j2 test-ci
+  - make lint
```
Comment (ljharb) [inline .travis.yml]: You can make it a separate task with `include`, an env var, and a conditional in `script` - it's probably worth doing it now.

**Comment by Fishrock123** [comment]
Oops, looks like the linter is still trying to build.

I really do not find travis' docs very usable. :/

File: tools/run-ci.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -ev
+
+# always change the working directory to the project's root directory
+cd $(dirname $0)/..
+
+if [ "${LINTER_ONLY}" = "true" ]; then
```
Comment (ljharb) [inline tools/run-ci.sh]: you may want `${LINTER_ONLY-}` just in case the `u` option is set

**Comment by sandfox** [comment]
Is the intention to have make `make lint` and `make test` run as separate (and therefore, parallelisable/concurrent as separate tasks)?

**Comment by Fishrock123** [comment]
> Is the intention to have make `make lint` and `make test` run as separate (and therefore, parallelisable/concurrent as separate tasks)?

Correct. This is how it is in the Node CI, e.g. https://ci.nodejs.org/job/node-test-commit/12497/.
It is helpful to have the linter results show separately.

File: tools/run-ci.sh (hunk 2 above)
Comment (Fishrock123) [reply tools/run-ci.sh]: I'm not sure what you mean - in case `set -u` is used in the bash script?

**Comment by TimothyGu** [comment]
Where is run-ci.sh used?

**Comment by Fishrock123** [comment]
@TimothyGu Please check the commit history. I have no way of testing this locally or marking this as "in progress".

Looks like this is working now though. I'm going to clean it up and squash it.

**Comment by Fishrock123** [comment]
Ok this _should_ be good to merge if this passes, I think.

**Review by TimothyGu — APPROVED** [review]

**Comment by TimothyGu** [comment]
I'm okay with landing this without a macOS run. Travis is have some issues with their macOS infrastructure: https://www.traviscistatus.com/

**Comment by Fishrock123** [comment]
Looks like it is building ok for us - its just taking a long tie to initially get started.

**Comment by Fishrock123** [comment]
All green.

**Comment by Fishrock123** [comment]
Can this be merged? 🤔

**Review by addaleax — APPROVED** [review]
yeah, landing

**Comment by addaleax** [comment]
Landed in 8e2de7752744182c29ea153b1c3f0100d87abd2a

**Pull request closed without merging**